dulwich/tests/compat/test_repository.py

```diff
@@ -119,3 +127,32 @@ def test_packed_objects(self):
     def test_all_objects(self):
         expected_shas = self._get_all_shas()
         self.assertShasMatch(expected_shas, iter(self._repo.object_store))
+
+
+@skipIf(git_version() < (2, 5), 'Git version must be >= 2.5')
+class WorkingTreeTestCase(ObjectStoreTestCase):
```

Comment:
`WorkingTreeTestCase` derives from `ObjectStoreTestCase` to re-test tests from `ObjectStoreTestCase` with a `Repo` that corresponds to a linked working tree. See the output from `python -m unittest -v dulwich.tests.compat_test_suite` (with the `-v` option):
```
test_all_objects (dulwich.tests.compat.test_repository.WorkingTreeTestCase) ... ok
test_bare (dulwich.tests.compat.test_repository.WorkingTreeTestCase) ... ok
test_head (dulwich.tests.compat.test_repository.WorkingTreeTestCase) ... ok
test_loose_objects (dulwich.tests.compat.test_repository.WorkingTreeTestCase) ... ok
test_packed_objects (dulwich.tests.compat.test_repository.WorkingTreeTestCase) ... ok
test_refs (dulwich.tests.compat.test_repository.WorkingTreeTestCase) ... ok
```